tools/cldr-code/src/main/java/org/unicode/cldr/test/ConsoleCheckCLDR.java

```diff
@@ -455,450 +479,503 @@ public static void main(String[] args) throws IOException {
-                }
-                final String parentID = LocaleIDParser.getParent(localeID);
-                if (parentID != null) {
-                    parent = cldrFactory.make(parentID, true);
-                }
-            } catch (RuntimeException e) {
-                fatalErrors.add(localeID);
-                System.out.println("FATAL ERROR: " + localeID);
-                e.printStackTrace(System.out);
-                continue;
-            }
-
-            // generate HTML if asked for
-            if (ErrorFile.generated_html_directory != null) {
-                String baseLanguage = localeIDParser.set(localeID).getLanguageScript();
+        // TODO: englishPaths doesn't seem to be used.
+        // final HashSet<String> ep = new HashSet<>();
+        // final CLDRFile displayFile = CheckCLDR.getDisplayInformation();
+        // addPrettyPaths(
+        //         displayFile, pathFilter, pathHeaderFactory, noaliases, true, ep);
+        // addPrettyPaths(
+        //         displayFile,
+        //         displayFile.getExtraPaths(),
+        //         pathFilter,
+        //         pathHeaderFactory,
+        //         noaliases,
+        //         true,
+        //         ep);
+        // final Set<String> englishPaths = Collections.unmodifiableSet(ep); // for robustness
+
+        // Set up our stream to use. It will be parallel usually, or sequential for HTML.
+        Stream<String> stream;
+
+        if (sequential) {
+            System.err.println("# Note: running in sequential mode.");
+            stream = locales.stream();
+        } else {
+            stream = locales.parallelStream();
+        }
 
-                if (!baseLanguage.equals(lastBaseLanguage)) {
-                    lastBaseLanguage = baseLanguage;
-                    ErrorFile.openErrorFile(localeID, baseLanguage);
-                }
-            }
+        // now, run it
+        stream.forEach(
+                localeID -> {
+                    if (ErrorFile.writeError != null) {
+                        return; // get out, it's an error.
+                    }
 
-            if (user != null) {
-                file = new CLDRFile.TestUser(file, user, isLanguageLocale);
-                if (parent != null) {
-                    parent = new CLDRFile.TestUser(parent, user, isLanguageLocale);
-                }
-            }
-            checkCldr.setCldrFileToCheck(file, new Options(options), result);
+                    List<CheckStatus> result = new ArrayList<>();
+                    Set<PathHeader> paths = new TreeSet<>(); // CLDRFile.ldmlComparator);
+                    Map<String, String> m = new TreeMap<>();
+                    Map<String, String> options = new HashMap<>();
+                    FlexibleDateFromCLDR fset = new FlexibleDateFromCLDR();
 
-            subtotalCount.clear();
+                    if (CLDRFile.isSupplementalName(localeID)) return;
```

Comment:
We should check to see if CLDRFile.isSupplementalName should be used anymore. It is a holdover from when the supplemental data files were in the same folder as the locale-based files (en.xml, etc).

Reply:
=> CLDR-17291